pyccl/tests/test_profiles.py

```diff
@@ -165,6 +165,34 @@ def test_cib_smoke():
         assert getattr(p, n) == 1234.
 
 
+def test_cib_2pt_diag():
+    c = ccl.halos.ConcentrationDuffy08(mass_def='200c')
+    p1 = ccl.halos.HaloProfileCIBShang12(concentration=c, nu_GHz=217,
+                                         mass_def='200c')
+    p2 = ccl.halos.HaloProfileCIBShang12(concentration=c, nu_GHz=190,
+                                         mass_def='200c')
+    p2pt = ccl.halos.Profile2ptCIB()
+
+    # Test diag=False
+    # TODO: I comment this out because the CIB_2pt returns np.array(number)
+    # which is not a float. Is this intended? This is not the case for
+    # Profile2pt.fourier_2pt
+    # F = p2pt.fourier_2pt(COSMO, 1., 1E13, 1., p1, prof2=p2,
+    #                    diag=False)
+    # assert isinstance(F, float)
```

Comment:
This is a weird numpy behaviour. The reason is that, in this case, `fourier_2pt` returns a squeezed array, which, even when it's a scalar, still has a `ndarray` type. This doesn't really matter, because it will actually behave as a scalar. In fact, when you take one of these squeezed arrays and multiply them by any other scalar (even another squeezed array), the result becomes a float.

So what's going on here, is that the standard `Profile2pt` gets the two squeezed profiles and multiplies them together (the result being a float). The CIB and HOD versions, however, return the squeezed array that their `fourier_variance` spits out, which is still a squeezed array.

As I said, none of this matters, since any of these squeezed arrays will become floats as soon as you do anything with them. So, instead, how about testing `assert np.ndim(F) == 0`?

Reply:
Done